driver-core/src/test/java/com/datastax/driver/core/CCMTestsSupport.java

```diff
@@ -283,31 +444,115 @@ private boolean dirtiesContext() {
             }
         }
 
+        private int weight() {
```

Comment:
The new weighing algorithm favors vanilla clusters over complex ones. 100% vanilla should weigh 0.